src/api/N2Go_Api.php

```diff
@@ -62,49 +62,42 @@ public static function run()
     private static function getPost($id)
     {
         global $wpdb;
+
+        /** @var WP_Post $post */
         $post = $wpdb->get_row(
             $wpdb->prepare("
                 SELECT 
-                    p.ID as ID, 
-                    p.guid as guid, 
-                    p.post_content as description,
-                    p.post_excerpt as shortDescription,
-                    p.post_title as title, 
-                    p.post_date as date,
-                    u.display_name as author 
+                    p.ID,
+                    p.post_content,
+                    p.post_excerpt,
+                    p.post_title, 
+                    p.post_date,
+                    u.display_name as post_author 
                 FROM $wpdb->posts p 
                     LEFT JOIN $wpdb->users u ON p.post_author = u.ID 
                 WHERE p.ID = %d
               ", $id)
         );
+        $result = null;
 
         if ($post) {
+            $basUrl = esc_url(home_url('/'));
+            $content = apply_filters('the_content', $post->post_content);
```

Comment:
**Note:** This is the standard way WordPress converts raw text into post content. Among other things it:
 converts smiley codes into images, converts symbols into nicer ones (eg, quotes, copyright etc), expands shortcodes into markup and replaces empty lines with break tags and *surrounds the content with a `<p>` tag*.

Reply:
sounds very good :)